Crash when depositing tokens

Using 6fdd7a12914a8601e2518511e2e8246c0b59e00b, when doing a batch of 150 deposits to Sapphire on Mainnet, I encountered the following crash. This occurred in 5-10% of deposits.

```
You are about to sign the following transaction:
Format: plain
Method: consensus.Deposit
Body:
  To: oasis1qrhq0ud0f7w70x445wkp760vdfz2agqcev3jl9r7
  Amount: 200.0 ROSE
Authorized signer(s):
  1. A2Mrg5uHLGnLkCErHgPtRJwvM8/w4lzGIZylfcWDrgI= (ed25519)
     Nonce: 76
Fee:
  Amount: 0.0 ROSE
  Gas limit: 11312
  (gas price: 0.0 ROSE per gas unit)

Network:  mainnet
ParaTime: sapphire
Account:  rtk
? Sign this transaction? Yes
(In case you are using a hardware-based signer you may need to confirm on device.)
Broadcasting transaction...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xc65081]

goroutine 42 [running]:
github.com/oasisprotocol/cli/cmd/common.WaitForEvent.func1()
        /home/oa/cli/cmd/common/transaction.go:440 +0x121
created by github.com/oasisprotocol/cli/cmd/common.WaitForEvent
        /home/oa/cli/cmd/common/transaction.go:431 +0x26c
```

--- a/cmd/common/transaction.go
+++ b/cmd/common/transaction.go
@@ -437,6 +437,10 @@ func WaitForEvent(
 			case <-ctx.Done():
 				return
 			case bev := <-ch:
+				if bev == nil {
+					return
+				}
+
 				for _, ev := range bev.Events {
 					if result := mapFn(ev); result != nil {
 						resultCh <- result

